crates/ruff/src/checkers/ast/analyze/bindings.rs

```diff
@@ -0,0 +1,68 @@
+use crate::checkers::ast::Checker;
+use crate::codes::Rule;
+use crate::rules::{flake8_import_conventions, flake8_pyi, pyflakes, pylint};
+use ruff_diagnostics::{Diagnostic, Fix};
+
+/// Run lint rules over the [`Binding`]s.
+pub(crate) fn bindings(checker: &mut Checker) {
+    if !checker.any_enabled(&[
+        Rule::InvalidAllFormat,
+        Rule::InvalidAllObject,
+        Rule::UnaliasedCollectionsAbcSetImport,
+        Rule::UnconventionalImportAlias,
+        Rule::UnusedVariable,
+    ]) {
+        return;
+    }
+
+    for binding in checker.semantic.bindings.iter() {
+        if checker.enabled(Rule::UnusedVariable) {
+            if binding.kind.is_bound_exception() && !binding.is_used() {
```

Comment:
Combine ifs

Reply:
I intentionally like to keep the `checker.enabled` checks separate from any preconditions for the rule. I find it easier to read, and I have a perhaps-mistaken belief that it might help with future refactors when we make the rule API more declarative.